File: apps/web/src/server/jobs/env.ts
/**
 * Configuration the job handlers need, read per delivery.
 *
 * Checked when a job runs rather than at boot — there is no long-lived worker
 * process anymore (ADR-012); the receiver endpoint is a serverless function,
 * so "startup" is every request. A missing variable must fail that delivery
 * loudly (500 → QStash retries → Sentry) instead of silently skipping.
 */

function required(name: string): string {
  const value = process.env[name]
  if (!value) {
    throw new Error(`${name} is not set`)
  }
  return value
}

export interface JobsEnv {
  telegramBotToken: string
  /** Public origin used to build every link in a message (no trailing slash). */
  appUrl: string
}

export function readJobsEnv(): JobsEnv {
  return {
    telegramBotToken: required('TELEGRAM_BOT_TOKEN'),
    appUrl: required('APP_URL').replace(/\/+$/, ''),
  }
}

File: apps/web/pkg/jobs/env.go
package jobs

import (
	"errors"
	"os"
	"strings"

	"countmein/pkg/logx"
)

// Configuration the job handlers need, read per delivery. Checked when
// a job runs rather than at boot — the receiver endpoint is a
// serverless function, so "startup" is every request. A missing
// variable must fail that delivery loudly (500 → QStash retries)
// instead of silently skipping.

type Env struct {
	TelegramBotToken string
	// Public origin used to build every link in a message (no trailing slash).
	AppURL string
}

func ReadEnv() (Env, error) {
	var e Env
	e.TelegramBotToken = strings.TrimSpace(os.Getenv("TELEGRAM_BOT_TOKEN"))
	e.AppURL = strings.TrimRight(strings.TrimSpace(os.Getenv("APP_URL")), "/")
	if e.TelegramBotToken == "" {
		logx.Info("TELEGRAM_BOT_TOKEN is not set", nil)
	}
	if e.AppURL == "" {
		logx.Info("APP_URL is not set", nil)
	}
	if e.TelegramBotToken == "" || e.AppURL == "" {
		return e, errors.New("jobs env is not configured (TELEGRAM_BOT_TOKEN / APP_URL)")
	}
	return e, nil
}
